1º passo: Carregar os dados

In [43]:
import pandas as pd

In [44]:
import numpy as np

In [45]:
dados_descontos = pd.read_csv('01.amazon_sales_dataset.csv')

In [46]:
dados_descontos.head()

,order_id,order_date,ship_date,delivery_date,order_status,customer_id,customer_name,country,state,city,...,product_name,category,sub_category,brand,quantity,unit_price,discount,shipping_cost,total_sales,payment_method
0,A10000,2026-01-31,2026-01-31,2026-01-08,Delivered,C5691,Ricky Potter,India,South Carolina,New Joe,...,without,Home,Furniture,Doyle-Jordan,3,42467.79,0.26,60.85,94339.3438,Card
1,A10001,2026-01-20,2026-02-03,2026-02-03,Delivered,C9811,Chris Davenport,India,Tennessee,Madisonmouth,...,school,Home,Furniture,Davis LLC,2,36138.76,0.24,112.96,55043.8752,COD
2,A10002,2026-01-15,2026-02-07,2026-01-03,Delivered,C7341,Timothy Gallagher,India,Iowa,East Larryberg,...,I,Electronics,Laptop,Watson and Sons,4,47148.93,0.14,64.11,162256.4292,UPI
3,A10003,2026-01-18,2026-01-15,2026-01-20,Delivered,C4012,Angela Collins,India,Kentucky,South Margaretshire,...,step,Electronics,Laptop,"Sandoval, Buckley and Solis",1,18487.99,0.06,114.15,17492.8606,Card
4,A10004,2026-01-27,2026-01-04,2026-01-23,Delivered,C1328,David Davidson DDS,India,North Dakota,Velasquezview,...,bit,Home,Kitchen,"Key, Bautista and Bowen",1,1742.25,0.10,40.47,1608.4950,UPI


In [47]:
dados_descontos.columns

Index(['order_id', 'order_date', 'ship_date', 'delivery_date', 'order_status',
       'customer_id', 'customer_name', 'country', 'state', 'city',
       'product_id', 'product_name', 'category', 'sub_category', 'brand',
       'quantity', 'unit_price', 'discount', 'shipping_cost', 'total_sales',
       'payment_method'],
      dtype='str')

2° passo: Isolar a coluna de desconto

In [48]:
descontos = dados_descontos['discount']


In [49]:
descontos = descontos.dropna()

3º passo : Medidas Centrais

In [50]:
# Calcular Media 

media = descontos.mean()
print('Média: ', media)

Média:  0.14996800000000002


In [51]:
# Calcular mediana(Q2)

mediana =  descontos.median()
print('Mediana: ', mediana)

Mediana:  0.15


4º passo: Medidas de Posição

In [52]:
# Calcular Máximo

maximo = descontos.max()
print('Máximo: ', maximo) 

Máximo:  0.3


In [53]:
# Calcular Mínimo 

minimo = descontos.min()
print('Mínimo: ', minimo)

Mínimo:  0.0


In [54]:
# Calcular Q1 (percentil 25)

q1 = np.percentile (descontos, 25)
print('Q1: ', q1)

Q1:  0.07


In [55]:
# Calcular Q3 (percentil 75)

q3 = np.percentile (descontos, 75)
print('Q3: ', q3)


Q3:  0.22


In [56]:
# Calcular AMPLITUDE 

amplitude = maximo -  minimo
print('Amplitude: ', amplitude)


Amplitude:  0.3


5º passo :  Assimetria

In [57]:
assimetria = descontos.skew()
print('Assimetria: ', assimetria)


Assimetria:  -0.004164539978905712



6º passo: Desafio de Interpretação

Como interpretar:

Se for ≈ 0 → distribuição simétrica

Se for > 0 → cauda à direita (muitos descontos baixos, poucos muito altos)

Se for < 0 → cauda à esquerda

In [58]:
# Maior e menor desconto

print(f'Maior desconto: $ {maximo:,.2f}')
print (f'Menor desconto: $ {minimo:,.2f}')

Maior desconto: $ 0.30
Menor desconto: $ 0.00


In [59]:
delta = media - mediana
print("Diferença (Média - Mediana):", delta)

Diferença (Média - Mediana): -3.199999999997649e-05


In [60]:
razao = media / mediana
print("Razão (Média/Mediana):", razao)

Razão (Média/Mediana): 0.9997866666666668


In [ ]:
###### Siga em baixo o relatorio descritivo em Markdown #######


1️º passo: Medidas Centrais

Média dos descontos: 0,149968 (~15%)

Mediana (Q2): 0,15 (15%)

 *Interpretação*

A média e a mediana são praticamente iguais.
A diferença entre elas é:

Delta (Média - Mediana): -0,000032

Razão (Média/Mediana): 0,9998

Isso indica que a distribuição dos descontos é extremamente equilibrada, sem influência significativa de valores extremos.

2º passo:  Medidas de Posição

Maior desconto: 0,30 (30%)

Menor desconto: 0,00 (0%)

Q1 (25%): 0,07 (7%)

Q3 (75%): 0,22 (22%)

Amplitude: 0,30

 *Interpretação*

25% dos descontos são menores que 7%

50% dos descontos estão entre 7% e 22%

75% dos descontos são menores que 22%

O desconto máximo aplicado foi de 30%

Existem produtos sem desconto (0%)

A amplitude de 30% mostra que existe variação, mas dentro de um limite controlado.

3º passo: Assimetria da Distribuição

Assimetria: -0,0042

 *Interpretação*

O valor da assimetria está extremamente próximo de zero.

Isso indica que a distribuição dos descontos é:

   *Praticamente simétrica
   *Bem equilibrada
   *Sem concentração anormal em valores altos ou baixos

Não há tendência significativa para descontos muito altos ou muito baixos.

 4º passo: Conclusão Geral

A análise estatística demonstra que:

A política de descontos é consistente

A média e mediana praticamente iguais indicam estabilidade

Não há presença relevante de outliers

A distribuição é equilibrada

A maioria dos descontos está concentrada entre 7% e 22%

O desconto máximo (30%) parece ser um teto controlado

 5º passo: Conclusão Final

A política de descontos da empresa demonstra padronização e controle, sem indícios de práticas agressivas ou distorcidas.
Os descontos estão bem distribuídos e seguem um comportamento estatisticamente estável.